javascript/extensible-search-typeahead.js

```diff
@@ -0,0 +1,229 @@
+;(function($) {
+	window.ESPPageStore = new PageStore;
+	$(window).load(function() {
+
+		$.fn.extend({
+			nearest: function (pattern) {
+
+				var found = null;
+				var domobj = null;
+
+				do {
+					domobj = this.parent();
+					found = domobj.find(pattern);
+				} while (found === null && domobj !== null)
+
+				return found;
+			}
+		});
+
+
+		var overlays = $('.esp-overlay');
+
+		if(overlays.length) {
+
+			bindSearchCapture();
+
+			overlays.each(function (i , overlay) {
+				overlay =  $(overlay);
+				var search = overlay.nearest('input.extensible-search.typeahead');
+				if(search.length) {
+
+					search.after(overlay);
+
+					if (ESPPageStore.getType('SearchTerms').length && overlay.children('.recent-searches').length) {
+
+						var resentsearch = overlay.find('.recentsearch').first().clone();
+						overlay.find('.recentsearch').remove();
+
+						overlay.find('.search-suggestions-list > li.list-item > a')
+							.each(function(){
+								$(this).attr('href',
+									'//' + window.location.host
+									+ $(this).parents('form').attr('action')
+									+ '?Search=' + $(this).text()
+								);
+							});
+
+						$.each(ESPPageStore.getType('SearchTerms'), function (i, term) {
+							overlay.find('.recent-searches-list').prepend(
+								resentsearch.clone()
+									.children('a')
+									.prop('href',term.link)
+									.text(term.title)
+									.parent()
+									.show()
+							);
+						});
+
+						$('.recent-searches').show();
+
+					}
+
+					search.keyup(function() {
+						var currSearch = $(this);
+						var data = {'page': currSearch.data('extensible-search-page')};
+						if(currSearch.val().length > 2) {
+							var list = overlay.find('.search-typeahead-list');
+							searchAjax(currSearch, data, list, 'extensible-search-api/getTypeahead');
+							var list = overlay.find('.search-suggestions-list');
+							searchAjax(currSearch, data, list, 'extensible-search-api/getSuggestions');
```

Comment:
Is it possible to create  single API call to perform these requests rather than making two separate ones? Much more efficient.